src/registry/customerProfile/subFields.tsx

```diff
@@ -0,0 +1,332 @@
+          </ButtonGroup>
+          <DigitalAddressPreferencesModal
+            isOpen={isModalOpen}
+            closeModal={() => setIsModalOpen(false)}
+            onSubmit={newPreference => {
+              // Update the preference fields.
+              setFieldValue(`${namePrefix}.isNewPreferred`, newPreference.isNewPreferred);
+              setFieldValue(`${namePrefix}.useOnlyOnce`, newPreference.useOnlyOnce);
+              setIsModalOpen(false);
+            }}
+            digitalAddressType={type}
+          />
+        </>
+      )}
+    </>
+  );
+};
+
+interface DigitalAddressTypeFieldProps {
+  namePrefix: string;
+  isRequired?: boolean;
+  digitalAddressGroup?: DigitalAddressGroup;
+}
+
+interface DigitalAddressFieldProps extends DigitalAddressTypeFieldProps {
+  type: DigitalAddressType;
+  textfieldProps: Partial<React.ComponentProps<typeof TextField>>;
+}
+
+/**
+ * A generic field for digital addresses. If there are no digital addresses yet, it shows
+ * a text input, otherwise it shows a select input with the addresses as options.
+ * The user can choose to add a new digital address, which replaces the select input for
+ * the text input.
+ *
+ * Only for new digital addresses can the user update their preferences. Updating the
+ * preference can be done in a modal, once the digital address textfield contains a valid
+ * digital address. In other situations, the user will have to use the online portal
+ * (outside of Open Forms) to update their preferences.
+ *
+ * @param namePrefix The name prefix for the fields.
+ * @param type The type of digital address field.
+ * @param isRequired Whether the fields are required.
+ * @param digitalAddressGroup The digital address group to display.
+ * @param textfieldProps Additional props to pass to the text input.
+ */
+const DigitalAddressField: React.FC<DigitalAddressFieldProps> = ({
+  namePrefix,
+  type,
+  isRequired,
+  digitalAddressGroup,
+  textfieldProps,
+}) => {
+  const id = useId();
+  const {setFieldValue, getFieldMeta} = useFormikContext();
+  const fieldName = `${namePrefix}.address`;
+
+  const {touched, error: formikError} = getFieldMeta(namePrefix);
+
+  // If there are no addresses yet, we show a text input.
+  const hasAddresses = !!digitalAddressGroup?.addresses?.length;
+  const [useTextInput, setUseTextInput] = useState(!hasAddresses);
+
+  const error = formikError as unknown as undefined | string | FormikErrors<DigitalAddress>;
+
+  const fieldError = typeof error === 'string' && error;
+  const invalid = touched && !!fieldError;
+
+  const errorMessageId = invalid ? `${id}-error-message` : undefined;
+
+  // When the digital addresses are loaded, we check if we need to show a text input.
+  useEffect(() => {
+    setUseTextInput(!hasAddresses);
+  }, [hasAddresses]);
+
+  return (
+    <Fieldset
+      isInvalid={invalid}
+      aria-describedby={errorMessageId}
+      className="openforms-customer-profile__digital-address-fieldset"
+    >
```

Comment:
This is a fieldset for only a single input field, no? Does this need to be a fieldset? It has no legend, no aria-describedby that's meaningful (except for the error message).